WordPress/Classes/ViewRelated/Blog/SiteManagement/SiteTagsViewController.swift

```diff
@@ -0,0 +1,151 @@
+import UIKit
+
+final class SiteTagsViewController: UITableViewController {
+    private struct TableConstants {
+        static let cellIdentifier = "TagsAdminCell"
+        static let numberOfSections = 1
+    }
+    private let blog: Blog
+    private let tagsService: PostTagService
+    private var tags: [PostTag] = []
+    
+    fileprivate let noResultsView = WPNoResultsView()
+    
+    @objc
+    public init(blog: Blog, tagsService: PostTagService) {
+        self.blog = blog
+        self.tagsService = tagsService
+        super.init(style: .grouped)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setAccessibilityIdentifier()
+        applyStyleGuide()
+        applyTitle()
+        initializeTable()
+        initializeNavigationBar()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        initializeData()
+    }
+    
+    private func setAccessibilityIdentifier() {
+        tableView.accessibilityIdentifier = "SiteTagsList"
+    }
+    
+    private func applyStyleGuide() {
+        WPStyleGuide.configureColors(for: view, andTableView: tableView)
+        WPStyleGuide.configureAutomaticHeightRows(for: tableView)
+    }
+    
+    private func applyTitle() {
+        title = NSLocalizedString("Tags", comment: "Label for the Tags Section in the Blog Settings")
+    }
+    
+    private func initializeTable() {
+        tableView.tableFooterView = UIView(frame: .zero)
+        tableView.register(WPTableViewCell.classForCoder(), forCellReuseIdentifier: TableConstants.cellIdentifier)
+        setupRefreshControl()
+    }
+    
+    private func setupRefreshControl() {
+        if refreshControl == nil {
+            refreshControl = UIRefreshControl()
+            refreshControl?.addTarget(self, action: #selector(refreshTags), for: .valueChanged)
+        }
+    }
+    
+    @objc
+    private func refreshTags() {
+        initializeData()
+    }
+    
+    private func initializeNavigationBar() {
+        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .add,
+                                                            target: self,
+                                                            action: #selector(createTag))
+    }
+    
+    @objc
+    private func createTag() {
+        let emptyTag = PostTag()
+        navigate(to: emptyTag)
+    }
+    
+    private func initializeData() {
```

Comment:
It looks like `refreshTags`, `initializeData` and `assign` are all involved in updating synced tags, and refreshing the table view.  `refreshTags` only calls `initializeData` whose main job appears to be syncing data. `assign` is responsible for updating the tableView but it looks like it gets called more than necessary -- if a user pulls to refresh, assign is called before there is new data.

It would be nice to consider:
- whether `refreshTags` is needed since it's mostly just  pass-thru to `initializeData`
- if there might be more descriptive names for intializeData (which handles syncing) and `assign` (which handles refreshing the table view).
- how to avoid calling `assign` when there's no changes to show.